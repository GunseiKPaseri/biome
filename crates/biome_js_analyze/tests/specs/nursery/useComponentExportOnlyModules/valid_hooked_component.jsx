import React, { memo } from 'react';

export const Component = () => <></>;
export const MemoArrowComponent = memo(() => <></>);
export const MemoNamedComponent = memo(Component);
export const MemoFunctionComponent = memo(function () {});
export const ReactMemoComponent = React.memo(() => <></>);
export const ReactMemoNamedComponent = React.memo(Component);
export const ForwardRefArrowComponent = forwardRef(() => <></>);
export const ForwardRefFunctionComponent = forwardRef(function () {});
export const ReactForwardRefComponent = React.forwardRef(() => <></>);
