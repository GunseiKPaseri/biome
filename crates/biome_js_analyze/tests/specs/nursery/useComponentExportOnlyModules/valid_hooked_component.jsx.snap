---
source: crates/biome_js_analyze/tests/spec_tests.rs
assertion_line: 86
expression: valid_hooked_component.jsx
---
# Input
```jsx
import React, { memo } from 'react';

export const Component = () => <></>;
export const MemoArrowComponent = memo(() => <></>);
export const MemoNamedComponent = memo(Component);
export const MemoFunctionComponent = memo(function () {});
export const ReactMemoComponent = React.memo(() => <></>);
export const ReactMemoNamedComponent = React.memo(Component);
export const ForwardRefArrowComponent = forwardRef(() => <></>);
export const ForwardRefFunctionComponent = forwardRef(function () {});
export const ReactForwardRefComponent = React.forwardRef(() => <></>);

```
